mono/eglib/glib.h

```diff
@@ -335,9 +335,9 @@ gchar*           g_win32_getlocale(void);
 /*
  * Precondition macros
  */
-#define g_warn_if_fail(x)  G_STMT_START { if (!(x)) { g_warning ("%s:%d: assertion '%s' failed", __FILE__, __LINE__, #x); } } G_STMT_END
-#define g_return_if_fail(x)  G_STMT_START { if (!(x)) { g_critical ("%s:%d: assertion '%s' failed", __FILE__, __LINE__, #x); return; } } G_STMT_END
-#define g_return_val_if_fail(x,e)  G_STMT_START { if (!(x)) { g_critical ("%s:%d: assertion '%s' failed", __FILE__, __LINE__, #x); return (e); } } G_STMT_END
+#define g_warn_if_fail(x)  G_STMT_START { if (!(x)) { g_warning ("%s:%d: assertion '%s' failed\n", __FILE__, __LINE__, #x); } } G_STMT_END
```

Comment:
All of these should be replaced by asserts, imho.

Reply:
I don't have any strong opinion there since I'm not familiar with how they're used in the code base; I just want output that isn't as annoying to look through for now.

Reply:
I think some should just be "softer" and quieter, not "harder", really.
Along the lines of `free(null)` just silently does nothing. Not worth a warning even.